### ISSUE TITLE
Contributor see a Token tab in the Address page

### ISSUE THREAD

**Issue by igorffs** [poanetwork/blockscout#410]
As a `Contributor`, I want to see a `Token` tab in the `Address Details` page, where I can see the list of `Tokens` the `Address` holds, as well as see the `Token Transfers` for each of those `Tokens`.

https://projects.invisionapp.com/share/8EMX9IOR2V9#/screens/308371747

## Acceptance Criteria
- Add a `Tokens` tab in the `Address Details` page, where there should be a list with all the `Tokens` that specific `Address` **has been involved in transfers**.
    - For each `Token`, there should be possible to see the following information:
        - `Token` name
            - If the token has no name, display the token `Address` or part of it.
        - Number of transfers (in which the associated `Address` has been involved)
        - The token type (ERC-20 or ERC-721)
    - There should be possible to paginate the `Tokens` list.

### Out of Scope
- This issue doesn't include the search field as shown in the mockup.
- This issue doesn't include the token icons as shown in the mockup.
- This issue doesn't include the filtered list of token transfers as shown in the mockup (see #561)
- This issue doesn't include the balances for the Tokens list (it depends on #514; see #564).

## Tasks

- [x] Create a query to get all tokens an address interacted with;
- [x] Build a query that counts the token transfers from the given address (maybe we can reuse the query above);
- [x] Create a controller to centralize the list of tokens given an address;
- [x] ~Refactor the token's list at the dropdown to use the new controller;~ (not needed anymore)
- [x] Create a new tab on the Address Details page;
- [x] Build the front-end;
  - [x] List the tokens that address has interacted with. Each token goes on its own card, with the following information:
  - [x] Token name, or token address if it has no name;
  - [x] Number of transfers that involved both that address and that token;
  - [x] Add pagination;
  - [x] Add links.

**Comment by gfreh** [comment]
- For each `Token`, there should be possible to see the following information:
    - `Token` balance (for the associated address) followed by the token symbol.
        - For `ERC20 Tokens`, the balance should be formatted as currency (thousands and decimal separators) considering the token's `decimals` info (if there is no `decimals` info, display the formatted amount). Example: `999,000.02 FREH`
        - For `ERC721 Tokens`, the balance of non-fungible tokens. Example: `803,843 CK`
        - In case the `Token` has no `symbol`, show nothing next to the amount.

**Comment by acravenho** [comment]
Do you think the number of transfers between that token and the account address should be included? I think this is a good metric to show.

**Comment by gfreh** [comment]
@acravenho do you mean in the Tokens tab? This is already in the acceptance criteria (it should be shown below the token name in the Token list).

**Comment by acravenho** [comment]
Perfect. I must have missed it.

**Comment by Lucasnar** [comment]
Besides the questions in the PR description, we thought of this one:

- If the address hasn't interacted with any tokens, should we display this tab? If yes, what should be inside it?

**Comment by acravenho** [comment]
@Lucasnar No, we should not display the tab.

**Comment by gfreh** [comment]
@acravenho two comments here: 

Firstly, we've split this issue in three as follows. This way, we believe we can reduce merge code faster and accelerate the feedback loop.
1. This one will handle displaying the tokens list with the number of transfers.
2. #561 will handle clicking on a token item and displaying the list of token transfers.
3.  #564 will handle displaying the token balances in the list (this also depends on #514).

Also, there are some other questions in the issue's first comment. Would you mind taking a look at them, please?

**Comment by acravenho** [comment]
@gfreh I'm not seeing the questions you're referring to.

**Comment by gfreh** [comment]
@acravenho I've listed the questions below:
- How should we sort the tokens list? We think a good option would be by the count of transfers.
- Should we differentiate ERC20 from ERC721 somehow in the list of tokens?

Closed the issue by mistake 😰 Sorry, @Lucasnar

**Comment by acravenho** [comment]
@gfreh 

1. We should only sort tokens that have a `balance` > `0`
2. I really like that idea of sorting by the number of transfers. Sounds perfect.
3. I believe we should list ERC-721 first and then list ERC-20 tokens. The other alternative is to create two columns.  ERC721 on the right and ERC20 on the left. This may be more difficult though.

**Comment by gfreh** [comment]
Hey @Lucasnar 

I believe everything is fine, except for the token type, which is not being shown. I'd suggest us to do something like `ERC-721    13 transfers` or something like that. I believe we've talked about that but I probably forgot to add this suggestion to the card.

After that, I think we could merge the PR.
